packages/core/src/storages/key_value_store.ts

```diff
@@ -202,8 +202,9 @@ export class KeyValueStore {
             validator: ow.isValid(k, ow.string.matches(KEY_VALUE_STORE_KEY_REGEX)),
             message: 'The "key" argument must be at most 256 characters long and only contain the following characters: a-zA-Z0-9!-_.\'()',
         })));
-        if (options.contentType && !ow.isValid(value, ow.any(ow.string, ow.buffer))) {
-            throw new ArgumentError('The "value" parameter must be a String or Buffer when "options.contentType" is specified.', this.setValue);
+        // @ts-expect-error `.pipe` duck typing
+        if (options.contentType && !(ow.isValid(value, ow.any(ow.string, ow.buffer)) || (ow.isValid(value, ow.object) && typeof value?.pipe === 'function'))) {
```

Comment:
`@ts-ignore/expect-error` should be rather last resort, better to cast just the problematic part and keep the type checks working for the rest of the line. i am ok with those in tests, but not so much in the actual codebase

```suggestion
        if (options.contentType && !(ow.isValid(value, ow.any(ow.string, ow.buffer)) || (ow.isValid(value, ow.object) && typeof (value as Dictionary)?.pipe === 'function'))) {
```